crates/ruff_python_formatter/src/expression/binary_like.rs

```diff
@@ -661,6 +661,14 @@ impl Format<PyFormatContext<'_>> for FlatBinaryExpressionSlice<'_> {
                     && is_simple_power_expression(
                         left.last_operand().expression(),
                         right.first_operand().expression(),
+                    )
+                    && !is_expression_parenthesized(
+                        left.last_operand().expression().into(),
+                        f.context().source(),
+                    )
+                    && !is_expression_parenthesized(
+                        right.first_operand().expression().into(),
+                        f.context().source(),
```

Comment:
I kept this out of `is_simple_power_expression` to keep that method `const` and trivia-agnostic, but not sure that's correct.

Reply:
I would prefer it in `is_simple_power_expression`. The loop here is already complicated enough... Keeping it `const` isn't that important as it is a private function anyway.